src/rpdk/core/data_loaders.py

```diff
@@ -368,6 +370,54 @@ def load_resource_spec(resource_spec_file):  # pylint: disable=R # noqa: C901
 
     return inlined
 
+def verify_listed_properties_are_defined(resource_spec):
+    LOG.debug("verify_listed_properties_are_defined")
+
+    schema_keys = [
```

Comment:
Some newer JSON pointer property types
```suggestion
    schema_keys = [
        "nonPublicProperties",
        "conditionalCreateOnlyProperties",
```
https://github.com/aws-cloudformation/cloudformation-resource-schema/blob/master/src/main/resources/schema/provider.definition.schema.v1.json